pkg/plugin/plugins.go

```diff
@@ -88,9 +88,21 @@ func (c *PluginsProcessor) Load() error {
 			if err != nil {
 				return fmt.Errorf("failed to load plugin %v", err)
 			}
-			c.Plugins = append(c.Plugins, plugins.([]v1.Plugin)...)
+			var plugs []v1.Plugin
+			for _, p := range plugins.([]v1.Plugin) {
+				for _, cp := range c.Plugins {
+					if p.Spec.Type == cp.Spec.Type && p.Spec.On == cp.Spec.On &&
```

Comment:
I suggest that we could encapsulate the judgement into a single function `func isSamePluginSpec(a,b v1.plugin) bool`.